toontown/leveleditor/DNASerializer.py

```diff
@@ -0,0 +1,265 @@
+            node = loadDNAFile(DNASTORE, Filename.fromOsSpecific(filename).cStr(), CSDefault, 1)
+        except Exception:
+            self.notify.debug(
+                    "Couldn't load specified DNA file. Please make sure storage code has been specified in Config.prc file")
+            return
+        if node.getNumParents() == 1:
+            # If the node already has a parent arc when it's loaded, we must
+            # be using the level editor and we want to preserve that arc.
+            newNPToplevel = NodePath(node)
+            newNPToplevel.reparentTo(hidden)
+        else:
+            # Otherwise, we should create a new arc for the node.
+            newNPToplevel = hidden.attachNewNode(node)
+
+        # Make sure the topmost file DNA object gets put under DNARoot
+        newDNAToplevel = self.editor.findDNANode(newNPToplevel)
+
+        # reset the landmark block number:
+        (self.landmarkBlock, needTraverse) = self.editor.findHighestLandmarkBlock(newDNAToplevel, newNPToplevel)
+
+        # Update toplevel variables
+        if needTraverse:
+            self.editor.createToplevel(newDNAToplevel)
+        else:
+            self.editor.createToplevel(newDNAToplevel, newNPToplevel)
+        # Create visible representations of all the paths and battle cells
+        self.editor.createSuitPaths()
+        self.editor.hideSuitPaths()
+        self.editor.createBattleCells()
+        self.editor.hideBattleCells()
+
+        self.editor.loadAnimatedProps(newNPToplevel)
+
+        # Set the title bar to have the filename to make it easier
+        # to remember what file you are working on
+        self.editor.panel["title"] = 'Open Level Editor: ' + os.path.basename(filename)
+        self.editor.panel.sceneGraphExplorer.update()
+        self.editor.popupNotification(f"Loaded {os.path.basename(filename)}")
+
+    def outputDNADefaultFile(self):
+        outputFile = self.outputFile
+        if outputFile == None:
+            self.saveToSpecifiedDNAFile()
+            return
+        file = os.path.join(dnaDirectory.toOsSpecific(), outputFile)
+        self.outputDNA(file)
+
+    def outputDNA(self, filename):
+        print('Saving DNA to: ', filename)
+        binaryFilename = Filename(filename)
+        binaryFilename.setBinary()
+        self.editor.DNAData.writeDna(binaryFilename, Notify.out(), DNASTORE)
+        self.editor.popupNotification(f"Saved to {os.path.basename(binaryFilename)}")
+        if ConfigVariableString("compiler") in ['libpandadna', 'clash']:
+            print(f"Compiling PDNA for {ConfigVariableString('compiler')}")
+            self.compileDNA(binaryFilename)
+
+    def compileDNA(self, filename):
+        from toontown.compiler.compile import process_single_file
+        process_single_file(filename)
+
+    def saveColor(self):
+        self.appendColorToColorPaletteFile(self.editor.panel.colorEntry.get())
+
+    def appendColorToColorPaletteFile(self, color):
+        obj = self.DNATarget
```

Comment:
Where does DNATarget come from? It isn't defined anywhere. Should it be base.le.DNATarget